go/roothash/api/commitment/pool.go

```diff
@@ -267,30 +239,111 @@ func (p *Pool) addOpenExecutorCommitment(
 		return ErrTxnSchedSigInvalid
 	}
 
-	// Check if the header refers to merkle roots in storage.
-	if uint64(len(body.StorageSignatures)) < p.Runtime.Storage.MinWriteReplication {
-		logger.Debug("executor commitment doesn't have enough storage receipts",
-			"node_id", id,
-			"min_write_replication", p.Runtime.Storage.MinWriteReplication,
-			"num_receipts", len(body.StorageSignatures),
-		)
-		return ErrBadStorageReceipts
-	}
-	if err := sv.VerifyCommitteeSignatures(scheduler.KindStorage, body.StorageSignatures); err != nil {
-		logger.Debug("executor commitment has bad storage receipt signers",
-			"node_id", id,
-			"err", err,
-		)
-		return err
-	}
-	if err := body.VerifyStorageReceiptSignatures(blk.Header.Namespace); err != nil {
-		logger.Debug("executor commitment has bad storage receipt signatures",
-			"node_id", id,
-			"err", err,
-		)
-		return p2pError.Permanent(err)
-	}
+	switch body.Failure {
+	case FailureNone:
+		// Ensure header fields are present.
+		if header.IORoot == nil {
+			logger.Debug("executor commitment is missing IORoot")
+			return ErrBadRootHashes
+		}
 
+		// Ensure header fields are present.
+		if header.StateRoot == nil {
```

Comment:
Maybe these checks should be moved into a `ValidateBasic() error` method of the `ComputeBody`? It would check that the message is internally consistent (e.g. if it indicates failure the fields are nil and otherwise they are non-nil). This would make it easier to check validity in various places if needed.

Reply:
done